src/common/transformations/src/transformations/op_conversions/convert_ti_to_sequences.cpp

```diff
@@ -830,6 +831,291 @@ ov::pass::ConvertLoopToLSTMSequence::ConvertLoopToLSTMSequence() {
     register_matcher(m, callback);
 }
 
+ov::pass::ConvertLoopWithSlicedInputConcatOutputToLSTMSequence::ConvertLoopWithSlicedInputConcatOutputToLSTMSequence() {
+    MATCHER_SCOPE(ConvertLoopWithSlicedInputConcatOutputToLSTMSequence);
+
+    auto loop_label = pattern::wrap_type<ov::op::v5::Loop>();
+
+    // pattern for condition sub-graph in Loop operarion
+    auto num_iter_const_label = pattern::wrap_type<op::v0::Constant>();
+    auto num_iter_param_label = pattern::wrap_type<op::v0::Parameter>();
+    auto num_iterations_label =
+        std::make_shared<pattern::op::Or>(OutputVector{num_iter_const_label, num_iter_param_label});
+    auto counter_label = pattern::wrap_type<op::v0::Parameter>();
+    auto counter_step_label = pattern::wrap_type<op::v0::Constant>();
+    auto updated_counter_label = pattern::wrap_type<op::v1::Add>({counter_label, counter_step_label});
+    auto less_label = pattern::wrap_type<op::v1::Less>({updated_counter_label, num_iterations_label});
+    auto condition_label = pattern::wrap_type<op::v0::Result>({less_label});
+
+    // pattern for LSTMCell in the body
+    auto xi_label = pattern::wrap_type<op::v0::Parameter>();
+    auto new_xi_shape_label = pattern::wrap_type<op::v0::Constant>();
+    auto xi_reshape_label = pattern::wrap_type<op::v1::Reshape>({xi_label, new_xi_shape_label});
+    auto init_hidden_state_i_label = pattern::wrap_type<op::v0::Parameter>();
+    auto init_cell_state_i_label = pattern::wrap_type<op::v0::Parameter>();
+    auto W_label = pattern::wrap_type<op::v0::Constant>();
+    auto R_label = pattern::wrap_type<op::v0::Constant>();
+    auto B_label = pattern::wrap_type<op::v0::Constant>();
+    auto lstm_cell_label = pattern::wrap_type<op::v4::LSTMCell>(
+        {xi_reshape_label, init_hidden_state_i_label, init_cell_state_i_label, W_label, R_label, B_label});
+    auto new_shape_hidden_state_label = pattern::wrap_type<op::v0::Constant>();
+    auto unsqueeze_hidden_state_label =
+        pattern::wrap_type<op::v1::Reshape>({lstm_cell_label, new_shape_hidden_state_label});
+    auto result_hidden_state_label = pattern::wrap_type<op::v0::Result>({unsqueeze_hidden_state_label});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
```

Comment:
This function looks too long. Is it possible to decompose it?

Reply:
Done